fetchai/ledger/serialisation/objects/transaction_api.py

```diff
@@ -81,3 +81,21 @@ def create_transfer_tx(address_from_bin, address_to_bin, amount, fee=0):
     tx.fee = fee
     tx.resources = [address_from_bin, address_to_bin]
     return tx
+
+
+def create_json_tx(contract_name, json_data, resources=None, raw_resources=None, fee=None):
+    resources = resources or []
+
+    # format the transaction
+    tx = Tx()
+    tx.contract_name = contract_name.encode()
+
+    if isinstance(json_data, dict):
```

Comment:
Bit ugly - this is needed to swich behaviours and not break api for people currently working on it.

Reply:
Yep fair enough we will need to revisit this in the future anyway